RUFAS/output_manager.py

```diff
@@ -0,0 +1,235 @@
+                                     'caller_function': info_map['caller_function']})
+        self.warnings_pool[key] = {'msg': msg, 'info_map': info_map}
+
+    def add_error(self, name: str, msg: str, info_map: Dict[str, Any]) -> None:
+        """
+        Adds an error message to the pool of errors.
+
+        Parameters
+        ----------
+        name : str
+            The name of the error
+        msg : str
+            The error message to be added to the pool
+        info_map: Dict[str, Any]
+            Additional args to be logged, some are non-optional
+        info_map['caller_class'] : str
+            The name of the class which called this function
+        info_map['caller_function'] : str
+            The name of the function which called this function 
+        """
+        key = self._generate_key(name,
+                                 {
+                                     'caller_class': info_map['caller_class'],
+                                     'caller_function': info_map['caller_function']})
+        self.errors_pool[key] = {'msg': msg, 'info_map': info_map}
+
+    def _generate_key(self, name: str,
+                      info_map: Dict[str, Union[str, bool]]) -> str:
+        """
+        Generates key for the pool.
+        See 'add_variable' docs for detailed arg description.
+
+        Raises
+        ------
+        KeyError
+            If either info_map['caller_class'] or info_map['caller_function'] are 
+            not present.
+        """
+        if info_map.get('caller_class') is None:
+            raise KeyError("'caller_class' were not found in info_map")
+        if info_map.get('caller_function') is None:
+            raise KeyError("'caller_function' were not found in info_map")
+
+        prefix = ""
+        if info_map.get('prefix') is not None:
+            prefix = info_map.get('prefix') + "."
+        elif not info_map.get('suppress_prefix', False):
+            prefix = self._get_prefix(info_map.get(
+                'caller_class'), info_map.get('caller_function')) + "."
+
+        suffix = ""
+        if info_map.get('suffix') is not None:
+            suffix = "." + info_map.get('suffix')
+        elif not info_map.get('suppress_suffix', False):
+            suffix = "." + self._get_suffix()
+
+        return f"{prefix}{name}{suffix}"
+
+    def _get_prefix(self, caller_class: str, caller_function: str) -> str:
+        """
+        Returns the prefix for a key in the pool.
+
+        Parameters
+        ----------
+        caller_class : str
+            The name of the class in which this key-value pair is originated
+        caller_function : str
+            The name of the function in which this key-value pair is originated
+
+        Returns
+        -------
+        str
+            {caller_class}.{caller_function}
+        """
+        return f"{caller_class}.{caller_function}"
+
+    def _get_suffix(self) -> str:
```

Comment:
perhaps
```suggestion
    def _get_time_suffix(self) -> str:
```
to prevent confusion related to the syntax differences with `_get_prefix()`?